service/history/handler.go

```diff
@@ -185,6 +185,9 @@ func (h *Handler) Start() {
 		h.config,
 	)
 	h.historyEventNotifier = events.NewNotifier(h.GetTimeSource(), h.GetMetricsClient(), h.config.GetShardID)
+	if h.config.EnableGracefulFailover() {
+		h.GetCoordinator().Start()
+	}
```

Comment:
I think this start logic should live in `resource` package as coordinator is part of history resource. Same for `Stop()`

Reply:
Updated